### PR TITLE
DOGE RPC Security Fix + addnode fragment + memory fixes for DOGE

### PR DESCRIPTION
**Pull request by 88plug**
DOGE is the only coin in BPS that uses a static rpcuser= and rpcpassword= that shouldn't be there.  
        rpcuser=ceiwHEbqWI83
        rpcpassword=DwubwWsoo3
is fair game for anyone to easily hack a new users node.

-Added unique rpcuser and rpcpassword generation for DOGE
-Removed static dbcache=50 so that the memory fragments will work properly (more-memory and save-memory)
-Addnodes fragment added using Prohashing explorer for eligable coins
-More memory and save memory now DOGE compatible
-Fixed spelling error

### CODE REVIEW

File: btcpay-setup.sh [hunk 1]
```diff
@@ -333,7 +333,14 @@ if [[ "$NBITCOIN_NETWORK" != "mainnet" ]] && [[ "$NBITCOIN_NETWORK" != "testnet"
     echo "NBITCOIN_NETWORK should be equal to mainnet, testnet or regtest"
 fi
 
-
+if [[ $BTCPAY_CRYPTOS == *"doge"* ]]; then
```
Comment (Kukks) [inline btcpay-setup.sh:336]: Please move this to a separate PR. I am unsure if there will be consensus around adding altcoin specific code in the main setup.

File: docker-compose-generator/docker-fragments/opt-addnodes.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        addnode=coins.prohashing.com:6001
+  viacoind:
+    environment:
+      BITCOIN_EXTRA_ARGS: |
+        addnode=coins.prohashing.com:6095
+  monacoind:
+    environment:
+      BITCOIN_EXTRA_ARGS: |
+        addnode=coins.prohashing.com:6138
+exclusive:
```
Comment (Kukks) [inline docker-compose-generator/docker-fragments/opt-addnodes.yml:25]: Remove this as there is no reason it would be exclusive since a node can have multiple addnode config items

File: docker-compose-generator/docker-fragments/opt-addnodes.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+version: "3"
```
Comment (Kukks) [inline docker-compose-generator/docker-fragments/opt-addnodes.yml:1]: You need to document this in the README.

File: docker-compose-generator/docker-fragments/opt-save-memory.yml [hunk 4]
```diff
@@ -16,6 +16,11 @@ services:
       BITCOIN_EXTRA_ARGS: |
         dbcache=50
         maxmempool=50
+  dogecoind:
+    environment:
+      DOGECOIN_EXTRA_ARGS: |
```
Comment (Kukks) [inline docker-compose-generator/docker-fragments/opt-save-memory.yml:21]: Why does the node support less memory all of a sudden?

**Review by Kukks — CHANGES_REQUESTED** [review]

File: btcpay-setup.sh (hunk 1 above)
Comment (88plug) [reply btcpay-setup.sh:336]: OK - will do - we need to do something though because static RPC user and pass will get users hacked eventually.

File: docker-compose-generator/docker-fragments/opt-addnodes.yml (hunk 3 above)
Comment (88plug) [reply docker-compose-generator/docker-fragments/opt-addnodes.yml:1]: OK

File: docker-compose-generator/docker-fragments/opt-addnodes.yml (hunk 2 above)
Comment (88plug) [reply docker-compose-generator/docker-fragments/opt-addnodes.yml:25]: Users should be able to add "addnodes" - depending on the use case.  Prohashing is a reliable source that runs seeder nodes, and I will be happy to document.

File: docker-compose-generator/docker-fragments/opt-save-memory.yml (hunk 4 above)
Comment (Kukks) [reply docker-compose-generator/docker-fragments/opt-save-memory.yml:21]: AFAIK save-memory for dogecoin was intentionally not introduced due to dogecoind needing all the ram

File: btcpay-setup.sh (hunk 1 above)
Comment (Kukks) [reply btcpay-setup.sh:336]: Not really, the dogecoin node is not exposed externally, it is firewalled inside the docker network. Many of the other services (lightning for example) have a hardcoded rpcauth in their yml file

File: docker-compose-generator/docker-fragments/opt-addnodes.yml (hunk 2 above)
Comment (Kukks) [reply docker-compose-generator/docker-fragments/opt-addnodes.yml:25]: I was referring to the `exclusive` section :)

File: docker-compose-generator/docker-fragments/opt-save-memory.yml (hunk 4 above)
Comment (88plug) [reply docker-compose-generator/docker-fragments/opt-save-memory.yml:21]: DOGE definetely does not need "all the ram" it needs a proper config though to manage it :)

File: docker-compose-generator/docker-fragments/opt-addnodes.yml (hunk 2 above)
Comment (88plug) [reply docker-compose-generator/docker-fragments/opt-addnodes.yml:25]: I thought exclusive was used when ignoring fragments, like this ```BTCPAYGEN_EXCLUDE_FRAGMENTS```
What is that section used for then?

File: btcpay-setup.sh (hunk 1 above)
Comment (88plug) [reply btcpay-setup.sh:336]: When DOGE is launched, Docker shows it is NOT using the "docker network"
```
0.0.0.0:22555
```
is exposed on the host as the rpcport!

Comment (Kukks) [reply btcpay-setup.sh:336]: That's only inside the docker container. It is not the actual host machine

File: docker-compose-generator/docker-fragments/opt-addnodes.yml (hunk 2 above)
Comment (Kukks) [reply docker-compose-generator/docker-fragments/opt-addnodes.yml:25]: It is used to avoid conflicting fragments, for example someone including a pruning fragment with the txindex fragment

**Comment by NicolasDorier** [comment]
We don't add alt specific settings to the setup. Fixed password are a problem we should fix, but this is not easy to do in a way that does not break existing installs.

**Comment by uhlhosting** [comment]
When will this be pushed to any release? I feel the high usage on Doge chain sync and how slow it runs alto we run on SSD.

**Comment by NicolasDorier** [comment]
Took the change from opt-*-memory on 51b700209b20babecb1114216dfa1b9b65e0284d kept the rest out.
